dashboard/src/components/SideMenu/SideMenu.tsx

```diff
@@ -46,47 +46,65 @@ const linkItems: LinkMenuItems[] = [
   },
 ];
 
-const emptyFunc = (): void => {};
+type SideMenuItemProps = {
+  item: RouteMenuItems;
+};
 
-const SideMenu = (): JSX.Element => {
+const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
   const selectedItemClassName =
     'w-full flex pl-5 py-4 cursor-pointer text-sky-500 bg-black border-l-4 border-sky-500';
   const notSelectedItemClassName =
     'w-full flex pl-5 py-4 cursor-pointer text-white';
 
-  const useNavigateTo = (path: string): (() => void) => {
-    const router = useRouter();
-    const { origin: unsafeOrigin } = useSearch({ strict: false });
+  const { pathname } = useLocation();
+
+  const isCurrentPath = pathname === item.navigateTo;
+
+  const router = useRouter();
+
+  const { origin: unsafeOrigin } = useSearch({ strict: false });
```

Comment:
A pattern that helps us not to break the hook rules is to aggregate all hooks at the beginning of the component